### PR TITLE
feat: homepage layout

### PR DESCRIPTION
**Pull request by daaniissh**
# **feat: Implemented Homepage and Component Enhancements**  

## Description  
This PR introduces the homepage, providing a feature-rich interface for browsing posts, along with improvements to various components.  

### Features  
- **Homepage Functionality:**  
  - Display of posts with upvotes, downvotes, comments, and user details.  
  - Infinite scroll for posts (if applicable).  

- **Component Enhancements:**  
  - Refactored components for better maintainability and performance.  
  - Added new icon components (`dots.svelte`, `upvote.svelte`, `downvote.svelte`).  
  - Removed unused components like `HomeHeader.svelte` and `formatNumber.ts`.  

- **Styling Updates:**  
  - Updated `app.css` and `tailwind.config.ts` for a consistent and modern design.  
  - Improved scrollbar styling with Tailwind's `scrollbar-thin` for a sleek look.  

- **Right Panel Improvements:**  
  - Refined `RightPost.svelte` and `rightPanel.svelte` for enhanced usability and appearance.  

### Additional Details  
- Introduced a new `Home` directory for organized component management.  
- Updated `+layout.svelte` and `+page.svelte` to integrate new features seamlessly.  

### How to Test  
1. Run the app locally.  
2. Navigate to the homepage and verify:  
   - Posts load correctly with accurate data.  
   - Interactions like upvotes, downvotes, and comments function as intended.  
   - Scrollbar styling and layout adjustments are visually appealing.  

### Screenshot
image

### Checklist  
- [x] Code is linted and formatted.  
- [x] No unused files or components remain.  
- [x] All new functionality has been thoroughly tested.  
- [x] Tailwind configurations are updated and applied consistently.  

## Additional Notes  
Please review the changes carefully, Feedback on the UI and functionality is welcome!

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -21,6 +21,7 @@ graph_*.png
 # Byte-compiled / optimized / DLL files
 *.py[cod]
 *$py.class
+/backend/.env.example
```
Comment (moonlitgrace) [inline backend/.gitignore:24]: what is your point on this?

File: frontend/src/lib/components/Icons/AboutIcon.svelte
Comment (moonlitgrace) [inline frontend/src/lib/components/Icons/AboutIcon.svelte:1]: seems like you havent updated your fork/local repo... i changed many stuff
ps: we're using an icon lib

File: frontend/src/lib/components/User/UserListItem.svelte [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<script lang="ts">
+	type User = {
+		name: string;
+		stared: boolean;
+		img: string;
+	};
+	import Star from '$lib/components/icons/StarIcon.svelte';
+
+	let { name, stared, img }: User = $props();
+	import StarFill from '$lib/components/icons/StarFill.svelte';
```
Comment (moonlitgrace) [inline frontend/src/lib/components/User/UserListItem.svelte:10]: imports must be on top

**Review by moonlitgrace — CHANGES_REQUESTED** [review]
@daaniissh i told you like thousand times to make a PR for whole homepage, not just leftpanel or rightpanel.

ps: consider these things next time.

File: frontend/src/routes/+layout.svelte [hunk 3]
```diff
@@ -13,5 +15,7 @@
 
 <main class="font-sans">
 	<Header />
+	<LeftPanel/>
+	<RightPanel/>
```
Comment (moonlitgrace) [inline frontend/src/routes/+layout.svelte:19]: right panel? what that component does?

**Review by moonlitgrace — CHANGES_REQUESTED** [review]
this branch is still not updated with upstream.

and please write tailwind classes a bit more carefully, wtf is `w-ful` and why do we need `cursor-pointer`? tailwind is already ugly, don't make it any worse.

running prettier could help.

File: frontend/src/lib/components/header.svelte [hunk 4]
```diff
@@ -50,7 +50,7 @@
 				<NotificationIcon class="size-6" />
 			</button>
 			<button aria-label="Profile" class="avatar btn size-10 overflow-hidden p-0">
-				<img draggable="false" alt="moonlitgrace" src="https://github.com/moonlitgrace.png" />
+				<img draggable="false" alt="moonlitgrace" src="https://i.pinimg.com/736x/ed/ee/a3/edeea3f5e941c09ab9542148aea64de4.jpg" />
```
[the remark this answers is not in the document]
Comment (daaniissh) [reply frontend/src/lib/components/header.svelte:53]: sorry

File: frontend/src/lib/utils/formatNumber.ts
[the remark this answers is not in the document]
Comment (daaniissh) [reply frontend/src/lib/utils/formatNumber.ts:1]: ok

File: frontend/src/lib/components/RightPanel/rightPanel.svelte [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<script>
+	import Post from './RightPost.svelte';
+</script>
+
+<div class="fixed right-0 h-screen w-80">
```
Comment (moonlitgrace) [inline frontend/src/lib/components/RightPanel/rightPanel.svelte:5]: not `fixed`, make a wrapper for right panel.
well check how reddit is doing this.

File: frontend/src/lib/components/Icons/AboutIcon.svelte
Comment (daaniissh) [reply frontend/src/lib/components/Icons/AboutIcon.svelte:1]: sorry , i forget that, i mean i didn't installed that package ,

[the remark this answers is not in the document]
Comment (daaniissh) [reply frontend/src/lib/components/Icons/AboutIcon.svelte:1]: not similar

File: frontend/src/lib/components/Icons/HelpIcon.svelte
[the remark this answers is not in the document]
Comment (daaniissh) [reply frontend/src/lib/components/Icons/HelpIcon.svelte:1]: done

**Review by moonlitgrace — APPROVED** [review]
LGTM! good work @daaniissh

**Pull request merged**